books/projects/shnf/top.lisp

```diff
@@ -318,19 +343,20 @@
         (norm-pow i (norm-add (list 'pow (- j i) r 0) p) (norm-add s q))))))
 
 (defun norm-add (x y)
-  (declare (xargs :measure (+ (shf-count x) (shf-count y))))
-  (and (shfp x)
-       (if (atom x)
-           (add-int x y)
-         (if (atom y)
-             (add-int y x)
-           (case (car x)
-             (pop (case (car y)
-                    (pop (add-pop-pop x y))
-                    (pow (add-pop-pow x y))))
-             (pow (case (car y)
-                    (pop (add-pop-pow y x))
-                    (pow (add-pow-pow x y)))))))))
+  (declare (xargs :measure (+ (shf-count x) (shf-count y))
```

Comment:
Remove `shfp` test. In program mode it follow from gurad. In logic mode is is not essential.